### ISSUE TITLE
Audit translation approver, and revoke translations by admins

### ISSUE THREAD

**Issue by Voileexperiments** [Codewars/codewars.com#1865]
Okay, so it's extremely easy to approve translations now since there's now a translation tab. But what to do if someone around just blindly accepts every single translation they can? Are there any ways to undo the inevitable damage they've done? Can we find out who approved the problematic translation in the first place?

Example: Someone approved the Haskell translation of [https://www.codewars.com/kata/59f7fc109f0e86d705000043](this kata), which is completely pointless because Haskell has native arbitrary sized integer support. It should be revoked.

**Comment by kazk** [comment]
The new pending translations tab is not shown to everyone. Only 20 users I explicitly granted can see it other than admins. I picked from users that was already approving translations and those that seemed to care about Codewars from past interactions/comments. I can revoke if they're abusing.

The user who approved is stored in the database so we can start to show it. Maybe on `/translations` page.

**Comment by kazk** [comment]
For undoing, it's currently not easy to do and it can cause data issues. I'm planning to add "soft" delete for comments so I can do similar for translations and prevent it from showing.

**Comment by Steffan153** [comment]
> The new pending translations tab is not shown to everyone.

What? When I click "Add New" it takes me there. Also, it says that closed translations are "visible only to moderators and authors", but I can see all the translations there (no, it's not my kata).

**Comment by anter69** [comment]
> > The new pending translations tab is not shown to everyone.
> 
> What? When I click "Add New" it takes me there. Or are you talking about another tab?

That's another thing. He is talking about this:
![image](https://user-images.githubusercontent.com/12795864/60773414-bbd94e80-a105-11e9-8639-d184dbe72abc.png)

**Comment by nomennescio** [comment]
A translation tab? That's great news! Finally we're heading towards a big next step in solving "how to get my translation approved" conundrum. Thanks!

**Comment by nomennescio** [comment]
As for technical issues for improving the quality of translation approvals:
- allow to see output for 'sample tests'
- allow to run tests for 'setup' code

**Comment by nomennescio** [comment]
@kazk any update on this already? I still have pending translations from 3 years ago. I don't feel the incentive to work on translations.

**Comment by FArekkusu** [comment]
>  I still have pending translations from 3 years ago. I don't feel the incentive to work on translations.

1. This issue is not about translations review/approval taking too long.
2. Content management is a voluntary work, and PUs are not **obliged** to review anything. If you want people to notice your pending translations, you can try bringing attention to them by leaving a comment. In fact, you did so, but...
3. Make sure that your translations are well-made before saying "review and **approve**" (or at least be ready to fix them). Today you bumped 4 blue/purple translations. I checked the 2 of them which I had access to, and rejected both because there were 6+ months old comments about issues which had never been fixed, and [here](https://www.codewars.com/kumite/5d121573eff3b8001decfb88?sel=5d121573eff3b8001decfb88) you even explicitly refused to fix anything - with such attitude I don't see how your "lack of incentive to translate anything" is bad...

**Comment by nomennescio** [comment]
> > I still have pending translations from 3 years ago. I don't feel the incentive to work on translations.
> 
> 1. This issue is not about translations review/approval taking too long.

I agree, but it helps in communicating dealing with pending translations, which is a longtime issue. It's therefore VERY difficult to get translations approved. 

> 2. Content management is a voluntary work, and PUs are not **obliged** to review anything. If you want people to notice your pending translations, you can try bringing attention to them by leaving a comment. In fact, you did so, but...

First of all: as if I don't know. Furthermore creating translations is voluntary work too, so what's your point? The point however is that pending translations often don't get noticed. For some Katas the author has longtime abandoned them. Hence my pending translations of 3 years old. This issue is about adding a translation tab which help communicating pending translations, which is great.

About your 3rd point, that is totally besides the point of this issue, I don't know why you bring that up. Still you seem to suggest that the issue might be about the quality of my translations, rather than a missing mechanism. I find your slur about the quality of my translations offensive. Contrary to what you're suggesting, the amount of my already approved translations, as well as the feedback I've got on them shows the general quality of them.

I don't mind getting feedback, but it does not mean all feedback by default is equally relevant.
The issue you explicitly refer to is a request to add random tests, which however are not in the original testcode either. I therefore think that's asking too much. The original requester did not respond to my reply anymore, so does not explicitly disagree with my reasoning.  That is not about explicitly refusing, just countering with an argument. If you think that is about an 'attitude', and hence you suggest I stop translating, well who are you to tell me that? As a matter of fact I've noticed in other issues that other commenters are actually having an issue with your attitude. Maybe it's time for some self-reflection.

As for the other translations, I've checked them, because I was not aware of any comments on them.
So instead of approving or leaving your remark, you decided to reject the 2 which had only some minor comments added, giving me a lot of unnecessary work to prepare these translation for review again. Have you made it your goal in life to annoy people? You're doing pretty well.
